client/package.json

```diff
@@ -16,7 +16,8 @@
     "lint-src": "eslint . --ext .js --ext .jsx --cache --ignore-pattern '**/__test__/**' --ignore-pattern 'coverage/**'",
     "lint-tests": "eslint . --ext .test.js --ext .test.jsx --cache -c .eslintrc.test",
     "lint": "yarn run lint-src && yarn run lint-tests",
-    "lint-fix": "yarn run lint-src --fix && yarn run lint-tests --fix"
+    "lint-fix": "yarn run lint-src --fix && yarn run lint-tests --fix",
+    "postinstall": "cd vendor/recorderjs && NODE_ENV=development yarn install"
```

Comment:
When should we run this command? do we need to run this in deployment?

Reply:
Because we added the recorderjs lib, we need to install dependency for it. Moreover, the lib added ```babel``` into the ```devDependencies``` hence we need to install in all the dependencies for it in development mode.

Reply:
hmm, so there's no need to run this in deployment?

Reply:
We have included ```recorderjs``` in our project. In order to run or compile our project, we also need to install all the dependences for ```recorderjs```. 
```postinstall``` will be executed after we run ```yarn install``` in the ```client``` folder of our project. The script in the ```postinstall``` is just to ensure we also install the dependencies for ```recorderjs``` when we want to install our own dependencies. 

The author of ```recorderjs``` included some dependencies of ```recorderjs``` in the ```devDependencies``` section , which will only be installed in the development environment. But actually some dependencies in the ```devDependencies```, such as ```babel``` is also needed the production environment for compilation. 

When ```Travis``` doing testing, it will only use production environment. It will run ```yarn install --production```. Let's assume if we do not specific the environment and only put ```cd vendor/recorderjs && yarn install``` for ```postinstall```, then the environment 
 to execute ```postinstall``` will follow the same environment as the command executed by ```Travis```, which is production environment. These dependencies in the ```devDependencies``` such as  ```babel``` for ```recorderjs``` will not be installed. But actually they are needed even in the production environment. Hence I added ```NODE_ENV=development``` to force install all the dependencies in ```recorderjs```'s ```package.json``` even for these dependencies in the ```devDependencies``` section. You may take a look at the ```package.json``` in the ```recorderjs``` folder.

Reply:
Thanks for the detailed explanation! Yes, I was indeed confused why `NODE_ENV=development` is required.

Reply:
Further did some experiment with this, the other way is to directly require the js file in dist folder. But interestingly the final bundle size is slightly smaller when we build it our own, I suspect it's because webpack's new treeshaking feature. So let's stick with this approach then 👍  Thanks @huangjosh !